packages/salesforcedx-vscode-core/src/commands/forceStartApexDebugLogging.ts

```diff
@@ -65,50 +65,41 @@ export class ForceStartApexDebugLoggingExecutor extends SfdxCommandletExecutor<{
           traceflag.DebugLevel.ApexCode,
           traceflag.DebugLevel.Visualforce
         );
-        developerLogTraceFlag.turnOnLogging();
-
-        const updateDebugLevelsCommandlet = new SfdxCommandlet(
-          new SfdxWorkspaceChecker(),
-          new EmptyParametersGatherer(),
-          new UpdateDebugLevelsExecutor()
-        );
-        const promises = [updateDebugLevelsCommandlet.run()];
+        await this.subExecute(new UpdateDebugLevelsExecutor().build());
 
-        let updateTraceFlagCommandlet;
         if (!developerLogTraceFlag.isValidDateLength()) {
           developerLogTraceFlag.validateDates();
-          updateTraceFlagCommandlet = new SfdxCommandlet(
-            new SfdxWorkspaceChecker(),
-            new EmptyParametersGatherer(),
-            new UpdateTraceFlagsExecutor()
-          );
+          await this.subExecute(new UpdateTraceFlagsExecutor().build());
         }
-        if (updateTraceFlagCommandlet) {
-          promises.push(updateTraceFlagCommandlet.run());
-        }
-
-        const updatePromise = Promise.all(promises);
-        await updatePromise;
       } else {
-        const createDebuglevelCommandlet = new SfdxCommandlet(
-          new SfdxWorkspaceChecker(),
-          new EmptyParametersGatherer(),
-          new CreateDebugLevel()
-        );
-        await createDebuglevelCommandlet.run();
+        resultJson = await this.subExecute(new CreateDebugLevel().build());
+        const debugLevelId = resultJson.result.id;
+        developerLogTraceFlag.setDebugLevelInfo(debugLevelId);
+
         const userId = await getUserId(
           vscode.workspace.workspaceFolders![0].uri.fsPath
         );
-        const createTraceFlagCommandlet = new SfdxCommandlet(
-          new SfdxWorkspaceChecker(),
-          new EmptyParametersGatherer(),
-          new CreateTraceFlag(userId)
-        );
-        await createTraceFlagCommandlet.run();
+        developerLogTraceFlag.createTraceFlagInfo();
+        resultJson = await this.subExecute(new CreateTraceFlag(userId).build());
+        developerLogTraceFlag.setTraceFlagId(resultJson.result.id);
       }
-      // run executors to update traceflag and debug levels
-      // tslint:disable-next-line:no-empty
-    } catch (e) {}
+      developerLogTraceFlag.turnOnLogging();
+      executionWrapper.successfulExit();
+    } catch (e) {
+      executionWrapper.failureExit(e);
+    }
+  }
+
+  public async subExecute(command: Command) {
```

Comment:
Maybe we should make this `private` first since I don't see it being used anywhere else.